fcomm/src/lib.rs

```diff
@@ -808,22 +808,48 @@ impl<'a> Proof<'a, S1> {
     }
 
     pub fn verify(&self, pp: &PublicParams) -> Result<VerificationResult, Error> {
-        let (public_inputs, public_outputs) = match self.claim {
-            Claim::Evaluation(_) => self.verify_evaluation(),
-            Claim::Opening(_) => self.verify_opening(),
+        let (public_inputs, public_outputs) = match &self.claim {
+            Claim::Evaluation(_) => self.evaluation_io(),
+            Claim::Opening(_) => self.opening_io(),
         }?;
 
-        let verified = self
-            .proof
-            .verify(pp, self.num_steps, public_inputs, &public_outputs)
-            .expect("error verifying");
+        let claim_iterations_and_num_steps_are_consistent = if let Claim::Evaluation(Evaluation {
+            iterations: Some(iterations),
+            ..
+        }) = self.claim
+        {
+            // Currently, claims created by fcomm don't include the iteration count. If they do, then it should be
```

Comment:
Please open an issue and link to the issue number here.

Reply:
done